src/Raven.Server/Rachis/Leader.cs

```diff
@@ -385,7 +385,15 @@ private void Run()
 
                 try
                 {
-                    _engine.SwitchToCandidateState("An error occurred during our leadership." + Environment.NewLine + e);
+                    ClusterTopology clusterTopology = null;
+                    using (_engine.ContextPool.AllocateOperationContext(out ClusterOperationContext context))
+                    using (var ctx = context.OpenReadTransaction())
+                    {
+                        clusterTopology = _engine.GetTopology(context);
+                    }
+
+                    if(clusterTopology!=null && clusterTopology.AllNodes.Count>1)
```

Comment:
1. formatting
2. let's add a new property `Count` to the `ClusterTopology` (similar to the `DatabaseTopology`)

Reply:
also, does it mean that the leader is dead? can we execute new command if we reach this code?